dnf/cli/commands/repoquery.py

```diff
@@ -340,55 +346,60 @@ def build_format_fn(self, opts, pkg):
             # there don't exist on the dnf Package object.
             raise dnf.exceptions.Error(str(e))
 
-    def _get_recursive_deps_query(self, query_in, query_select, done=None, recursive=False,
-                                  all_deps=False):
-        done = done if done else self.base.sack.query().filterm(empty=True)
-        t = self.base.sack.query().filterm(empty=True)
-        set_requires = set()
-        set_all_deps = set()
-
-        for pkg in query_select.run():
-            pkg_provides = pkg.provides
-            set_requires.update(pkg_provides)
-            set_requires.update(pkg.files)
```

Comment:
I believe that your new code does not add pkg.provides in the pool for requires search. Am I correct?

Reply:
This is what the new libsolv function `pool_whatmatchessolvable()` does, see https://github.com/rpm-software-management/libdnf/pull/762. What I'm doing now is resolving the NEVRAs to packages, which is the first half of the removed code you're referencing does, and then pass the packages to the Query filter, which calls the libsolv function on them.

Reply:
The libsolv function should work for file provides as well. We've got one test for a file provide: https://github.com/rpm-software-management/ci-dnf-stack/blob/06b09749550cb27e20d450d5b89a77e374ce1815/dnf-behave-tests/features/repoquery/deps.feature#L258

Which tests the `--whatrequires` switch and therefore covers this. The other switches go through the same code path, so I think this is covered, but we could possibly add tests for file provides for some more switches (it's just a bit complex adding this to various packages, you either influence other tests or need more test packages and it makes the test data more confusing either way).